common/src/main/java/com/microsoft/identity/common/CodeMarkerManager.java

```diff
@@ -0,0 +1,111 @@
+//  Copyright (c) Microsoft Corporation.
+//  All rights reserved.
+//
+//  This code is licensed under the MIT License.
+//
+//  Permission is hereby granted, free of charge, to any person obtaining a copy
+//  of this software and associated documentation files(the "Software"), to deal
+//  in the Software without restriction, including without limitation the rights
+//  to use, copy, modify, merge, publish, distribute, sublicense, and / or sell
+//  copies of the Software, and to permit persons to whom the Software is
+//  furnished to do so, subject to the following conditions :
+//
+//  The above copyright notice and this permission notice shall be included in
+//  all copies or substantial portions of the Software.
+//
+//  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+//  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+//  FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+//  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+//  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+//  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+//  THE SOFTWARE.
+package com.microsoft.identity.common;
+
+import java.text.SimpleDateFormat;
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.List;
+
+/**
+ * A utility which collects any event's information and also provides functionality to retrieve the events in csv format.
+ */
+public class CodeMarkerManager {
+
+    private boolean enableCodeMarker = false;
+    // MAX_SIZE_CODE_MARKER is the maximum number of markers this utility can have.
+    private static final int MAX_SIZE_CODE_MARKER = 1000;
+    private volatile List<CodeMarker> codeMarkers = new ArrayList<>();
+
+    //baseMilliSeconds is the time in milliseconds when first codemarker was captured.
+    private long baseMilliSeconds = 0;
+    private String scenarioCode = null;
+    private static CodeMarkerManager sCodeMarkerManager = new CodeMarkerManager();
+
+    public static CodeMarkerManager getInstance() {
+        return CodeMarkerManager.sCodeMarkerManager;
+    }
+
+    private CodeMarkerManager() {
+    }
+
+    /**
+     * This method sets a scenarioCode Defined in {@link PerfConstants.ScenarioConstants}.
+     * This scenario code will be pre-fixed to every code marker.
+     * @param scenarioCode
+     */
+    public void setPrefixScenarioCode(String scenarioCode) {
+        this.scenarioCode = scenarioCode;
```

Comment:
Same comment here about var naming, if we use a member prefix, we can avoid sprinkling `this` about the code